### ISSUE TITLE
Getting warning saying Grid, Row and Col are deprecated

### ISSUE THREAD

**Issue by sunilkumarc** [react-native-training/react-native-elements#519]
I'm trying to use Grid, Row and Col components in my expo application. But I'm getting warning saying these components are deprecated and will be removed in future version of react native elements. What should I do?

**Comment by iRoachie** [comment]
Hey @sunilkumarc it means we are no longer supporting those components and they're going to be removed soon form the library. Grid, Row and Col were only `<View />` components that had a `flexDirection` styling. We initially added them for new beginners starting with the library. However now we think that persons using react native should learn and get familiar with flex box and styling.

Here's a writeup on how flexBox works in react native. https://facebook.github.io/react-native/docs/flexbox.html

**Comment by sunilkumarc** [comment]
Makes sense. Thanks for the information.

**Comment by nfplay** [comment]
Hi @iRoachie , thank you for your work on react-native-elements, it's a stellar module, full of functionalities.

I must say i was a bit surprised about Grid, Row and Col deprecation. Normally functionalities get deprecated because they no longer serve their purpose well, or a better alternative has come up. In this case, it is being abandoned because people should learn how it is made? I find this a bit weird. It is always better to be familiar with how the things are done under the hood, but that defies the point of creating modules in the first place. IMO, that decision should be made by the developer, not enforced by the author. The Grid still being available in the project's page doesn't help either.

Don't get me wrong, i find react-native-elements a fantastic module and i thank all the contributors for all the hard work that has been put to it, but functionalities are not tutorials. Just my 2 cents.

Thanks,

**Comment by iRoachie** [comment]
@nfplay Thanks for your words. I'm actually one of the newest maintainers here, and the decision to deprecate those components was made before I came.

Flexbox is at the heart of react native interfaces, and you'll probably use it even to build the simplest of apps. Grid, Row, and Col are very simple components built around flexbox.

**Grid**
Grid is just a `<View>` with style of `flexDirection: 'row'` or `flexDirection: 'column`

**Col and Row**
Both are just `<View>` with a style of `{{ flex: number }}`.

All were just very beginner abstractions, and don't actually provide much value to the developer. Thanks for noting about them being still in the docs, they'll be removed

**Comment by Monte9** [comment]
@nfplay thanks for the detailed feedback. 

@iRoachie explained it very well. `All were just very beginner abstractions, and don't actually provide much value to the developer.` In the case of Grid/Col/Row components, the abstractions are so trivial that it's almost easier to have the developer start using flexbox instead of relying on something like the Grid Component. 

Here's an example: https://github.com/react-native-training/react-native-elements/commit/5f589d76e2de02804aaabebbef8b44bc0fbfbdef#diff-ace61999bdf81cf330968627970eb5a2L330

So basically, the 
- `Grid` is just a `<View style={{flex: 1}}` 
- `Row` is just a `<View syle={{flexDirection: 'row'}}`
- `Col` is just a `<View style={{flexDirection: 'column'}}`

Hope this explanation helps anyone wanting to migration from Grid -> flexbox

**Comment by dx51job** [comment]
Thanks! Got it ^_^

**Comment by Oxyaction** [comment]
Suggest updating the docs with deprecated mark. Getting deprecation warn currently is pretty confusing.

**Comment by iRoachie** [comment]
It’s on the checklist for 0.17 @Oxyaction

**Comment by theurbanforrest** [comment]
I was originally a little grumpy about these being taken away.  then i spent 30 minutes jumping into flexbox tutorials and am happy you forced me to learn it :)  great stuff team

**Comment by yordis** [comment]
I would say that if you really want to teach the community then just add those simple lines of code explaining what it does.

I would encourage to  use for a simple fact that is more declarative of your layout and I don't have to be creating `{ wrapper: { flex: 2 }, anotherWrapper: { 1 } }` styles with some `View` and just go with the declarative style using the components (less code, same outcome). And also you could extend it to add some other flexbox styles like alignment and stuff like that.

I definitely do not agree with this change, like @nfplay pointed out, is up to the developer to use or not specific feature.

And once again, if you want to teach the community then write a readme encourage people to look at the source code and see how simple it is.

I hope you change your mind.
